### ISSUE TITLE
Add SeqSet

### ISSUE THREAD

**Issue by NthPortal** [scala/scala-library-next#22]
Add `SeqSet`, mirroring `SeqMap`.

**Comment by NthPortal** [comment]
Note: without adding new implementations with the traits, this depends on #35

**Comment by NthPortal** [comment]
everyone okay with this change? (👍 in favour, 👎 against)

**Comment by julienrf** [comment]
So, we would have `LinkedHashSet` as a concrete subtype of `SeqSet`?

**Comment by NthPortal** [comment]
yes, although it would take some fiddling to make that actually work in scala-library-next (if it's not final, we might be able to subtype it with a mixin?)

**Comment by SethTisue** [comment]
We'd want `VectorSet` as well: #168
